honeybee/model.py

```diff
@@ -22,76 +25,235 @@ class Model(object):
+                 orphaned_apertures=[], orphaned_doors=[], north_angle=0):
         """A collection of Rooms, Faces, Apertures, and Doors for an entire model.
 
         Args:
             name: Model name. Must be < 100 characters.
-            objects: A list of honeybee Rooms, Faces, Apertures, and Doors.
+            rooms: A list of Room objects in the model.
+            orphaned_faces: A list of the Face objects in the model that lack
+                a parent Room. Note that orphaned Faces are not acceptable for
+                Models that are to be exported for energy simulation.
+            orphaned_shades: A list of the Shade objects in the model that lack
+                a parent.
+            orphaned_apertures: A list of the Aperture objects in the model that lack
+                a parent Face. Note that orphaned Apertures are not acceptable for
+                Models that are to be exported for energy simulation.
+            orphaned_doors: A list of the Door objects in the model that lack
+                a parent Face. Note that orphaned Doors are not acceptable for
+                Models that are to be exported for energy simulation.
             north_angle: An number between 0 and 360 to set the clockwise north
                 direction in degrees. Default is 0.
         """
-        self._name = valid_string(name, 'honeybee model name')
-        self._name_original = name
+        self.name = name
         self.north_angle = north_angle
 
         self._rooms = []
         self._faces = []
         self._shades = []
         self._apertures = []
         self._doors = []
-        self._room_names = []
-        self._face_names = []
-        self._shade_names = []
-        self._aperture_names = []
-        self._door_names = []
+        for room in rooms:
+            self.add_room(room)
+        for face in orphaned_faces:
+            self.add_face(face)
+        for shade in orphaned_shades:
+            self.add_shade(shade)
+        for aperture in orphaned_apertures:
+            self.add_aperture(aperture)
+        for door in orphaned_doors:
+            self.add_door(door)
+
+        self._properties = ModelProperties(self)
+
+    @classmethod
+    def from_dict(cls, data):
+        """Initialize a Model from a dictionary.
+
+        Args:
+            data: A dictionary representation of a Model object.
+        """
+        # check the type of dictionary
+        assert data['type'] == 'Model', 'Expected Model dictionary. ' \
+            'Got {}.'.format(data['type'])
+
+        # import apertures
+        child_apertures = {}
+        orphaned_apertures = []
+        if 'apertures' in data:
+            for ap in data['apertures']:
+                ap_type_class = getattr(hat, ap['aperture_type']['type'])
+                ap_type = ap_type_class.from_dict(ap['aperture_type'])
+                if not ap['boundary_condition']['type'] == 'Surface':
```

Comment:
I can't follow the logic here. Is it supposed to be `if ap['boundary_condition']['type'] == 'Surface':`? now it reads as if it is not surface then set it to outdoors.

Reply:
At the moment, Apertures and Doors can only have one of two possible boundary conditions: `Outdoors` and `Surface`.  `Ground` boundary conditions for apertures are illegal as are `Adiabatic` boundary conditions.  If you want, I could just rewrite this statement as:
```
if ap['boundary_condition']['type'] == 'Outdoors':
```
... and it would do the same thing. I was only trying to anticipate a future scenario where we might support another type of boundary condition when I wrote it this way but, now that I think about it, I don't think we will ever support another option for apertures other than those two.  So I can chage it if that's desired.

Reply:
This is all now completely re-written.